pkg/frontend/reader/reader.go

```diff
@@ -200,3 +215,24 @@ func extractDisplayMode(cacheId model.TraceID) tfconfig.Id {
 	displayMode := cacheId.High >> CacheIdHighBitShift
 	return tfconfig.Id(uint32(displayMode))
 }
+
+func filterTimeRange(spans []*model.Span, startTime, endTime time.Time) []*model.Span {
+	var retained []*model.Span
+
+	for _, span := range spans {
+		traceSource, exists := model.KeyValues(span.Tags).FindByKey(zconstants.TraceSource)
+		if exists && traceSource.VStr == "object" {
+			// pseudo span, timestamp has no meaning
+			span.StartTime = startTime
+			span.Duration = endTime.Sub(startTime)
```

Comment:
if the search time range is longer than half hour,  but all children spans are within half hour, the object span displayed duration may be  a little weird ?

Reply:
@xuqingyun It would still split, it just no longer displays the whole half hour, but it is more clear this way.

And I am going to fix #14 next week anyway so it shouldn't be a problem.

Reply:
I'm just splitting down the feature to smaller commits; we don't need to merge this yet.

Reply:
Your concern is resolved in #55.